libtextsecure/message_receiver.js

```diff
@@ -1282,19 +1348,42 @@ MessageReceiver.prototype.extend({
   async handleEndSession(number) {
     window.log.info('got end session');
     const deviceIds = await textsecure.storage.protocol.getDeviceIds(number);
+    const identityKey = StringView.hexToArrayBuffer(number);
+    let conversation;
+    try {
+      conversation = window.ConversationController.get(number);
+    } catch (e) {
+      window.log.error('Error getting conversation: ', number);
+    }
 
-    return Promise.all(
-      deviceIds.map(deviceId => {
+    // Bail early if a session reset is already ongoing
+    if (conversation.isSessionResetOngoing()) {
+      return;
+    }
```

Comment:
Do we want to commit to always doing on liner blocks like this without braces?

Reply:
I'm ok with how it currently is, as it's the existing style adopted by Signal.